lib/config.go

```diff
@@ -459,7 +462,19 @@ func (c *RuntimeConfig) Validate(onExecution bool) error {
 
 	// check we do have at least a runtime
 	if _, ok := c.Runtimes[c.DefaultRuntime]; !ok {
-		return errors.New("no default runtime configured")
+		// Set the default runtime to "runc" if default_runtime is not set
+		if c.DefaultRuntime == "" {
+			logrus.Warnf("Defaulting to %q as the runtime since default_runtime is not set", defaultRuntime)
```

Comment:
Why is this a warning?  This is a *completely normal* condition where we may have an older config file right?

Upgrading is a normal thing :wink:

Reply:
I don't want to be a pain - but I really think it's worth emphasizing - we have to care *deeply* about upgrades.  It does mean "carrying the weight" of previous design decisions; files that get laid down on disk, particularly config files that admins might change are things that we need to avoid breaking *years* into the future.

I don't know the full backstory here and I may be missing something, but it looks to me like we're going out and emitting a message into the logs because a human (or code) didn't go by and tell cri-o "please use `runc`".  But the use case for changing the runtime seems sufficiently obscure (hm well I guess there's kata and stuff) that again it seems totally normal to me to just silently default to `runc`.

Reply:
Would making it a `debug` be better? I am thinking of the cases where users don't realize that they have to set `default_runtime` to tell cri-o to use a runtime apart from runc.

Reply:
Switched it to be a debug message instead.